### DISCUSSION TITLE
Use of records in the standard library

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#9038] [General]
We have generally refrained from using records in the library.

The only place they have been used are for functions with multiple arguments of the same type.

As we turn towards effects, the question is whether we should use them more.

Consider:

```scala
type alias HttpResponse = {status = Int32, statusText = String, url = String, body = String}

eff Http {
    pub def get(url: String, headers: Map[String, String]): HttpResponse
}
```

The nice thing is that `HttpResponse` can be extended with additional fields without breakage.

On the other hand, we can also introduce a type `HttpResponse` and then `def getBody(r: HttpResponse): String` and then there is no breakage either-- but usage may be harder.

**Comment by magnus-madsen** [comment]
@JonathanStarup @mlutze @jaschdoc @stephentetley I would like to know your thoughts

**Comment by mlutze** [comment]
> The nice thing is that HttpResponse can be extended with additional fields without breakage.

This is only true if one doesn't pattern-match on it.

**Reply by jaschdoc** [reply]
Not if you allow the pattern to have more labels e.g., `case { label1 = x, label2 | _ } => ???`

**Comment by JonathanStarup** [comment]
This is a behaviour def. vs data def issue right? I think both are valid so we need to judge them case by case. I'd say httpresponse is very much data so a record makes sense
